### ISSUE TITLE
[no-dupe-class-members] false negative for method overload

### ISSUE THREAD

**Issue by teppeis** [typescript-eslint/typescript-eslint#291]
**Repro**

```JSON
{
  "rules": {
    "no-dupe-class-members": "error"
  }
}
```

```TS
class Foo {
  foo(a: string): string;
  foo(a: number): number;
  foo(a: any): any {}
}
```

**Expected Result**
no errors

**Actual Result**
```
  29:3  error  Duplicate name 'foo'  no-dupe-class-members
  30:3  error  Duplicate name 'foo'  no-dupe-class-members
```

**Additional Info**

[no-dupe-class-members](https://eslint.org/docs/rules/no-dupe-class-members) is a recommended rule in eslint, so many users enable it.

My suggestions:

A. disable in `plugin:@typescript-eslint/recommended`
B. create `@typescript-eslint/no-dupe-class-members` and add an option to allow overload in TypeScript
C. notice in document

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.3.0` |
| `@typescript-eslint/parser`        | `1.3.0` |
| `TypeScript`                       | `3.3.3` |
| `ESLint`                           | `5.14.0` |
| `node`                             | `10.15.1` |
| `npm`                              | `6.8.0` |

**Comment by teppeis** [comment]
`tsc` throws an error for duplicate methods without ESLint.

```ts
class Foo {
    bar() {}
    bar() {} // Error: Duplicate function implementation.
}
```

So [no-dupe-class-members] may not be necessary.

**Comment by mysticatea** [comment]
Thank you for the report.

I think that we should have the setting to disable `no-dupe-class-members` rule in our recommended preset.

```json
{
    "overrides": [
        {
            "files": ["*.ts", "*.tsx"],
            "rules": {
                "no-dupe-class-members": "off"
            }
        }
    ]
}
```

And we should not have any alternative rule since `tsc` checks it.

**Comment by brainkim** [comment]
I’m not fond of the idea of merely relying on tsc compiler errors. Sometimes I’m using babel to compile typescript and it would be nice to be notified of dupe class members anyways.

**Comment by WORMSS** [comment]
> `tsc` throws an error for duplicate methods without ESLint.
> 
> ```ts
> class Foo {
>     bar() {}
>     bar() {} // Error: Duplicate function implementation.
> }
> ```
> 
> So `no-dupe-class-members` may not be necessary.

Just to note, your test is not the same as OP 

You have actually wrote two implementations, where as the OP as an single implementations with 2 overrides.
